api/src/services/artifact-service.ts

```diff
@@ -144,4 +146,45 @@ export class ArtifactService extends DBService {
 
     await this.artifactRepository.updateArtifactSecurityReviewTimestamp(artifactId);
   }
+
+  /**
+   * Deletes multiple artifacts and their related S3 objects for a given list of UUIDs
+   *
+   * @param {string[]} uuids UUIDs of artifacts to delete
+   */
+  async deleteArtifacts(uuids: string[]): Promise<void> {
+    defaultLog.debug({ label: 'deleteArtifacts' });
+
+    try {
+      for (const uuid of uuids) {
+        await this.deleteArtifact(uuid);
+      }
+    } catch (error) {
+      throw new ApiGeneralError(`There was an issue deleting an artifact.`);
+    }
+  }
+
+  /**
+   * Deletes an artifact and related S3 object for a given UUID
+   *
+   * @param {string} uuid UUID of artifact to delete
+   */
+  async deleteArtifact(uuid: string): Promise<void> {
+    defaultLog.debug({ label: 'deleteArtifact' });
+
+    const artifact = await this.artifactRepository.getArtifactByUUID(uuid);
+
+    if (artifact) {
+      try {
+        const service = new SecurityRepository(this.connection);
+        await service.deleteSecurityRulesForArtifactUUID(uuid);
+
+        await this.artifactRepository.deleteArtifactByUUID(uuid);
```

Comment:
You could also just call the `deleteArtifacts([uuid])` function above. Not sure which is better, or if it really matters tbh.

Reply:
there is only a single repo function for deleting an artifact, I didn't create a batch delete in the repo because of the S3 object delete

Reply:
I was thinking the service function above, that iterates over an array of uuids. But in hindsight, theres really no point int calling it over the repo function. So ignore this!